pkg/server/node.go

```diff
@@ -451,25 +451,32 @@ func (n *Node) start(
 		}
 		return nil
 	}); err != nil {
-		return errors.Wrapf(err, "failed to read last up timestamp from stores")
+		return errors.Wrapf(err, "failed to read last up timestamp from stores"), nil
 	}
 	n.lastUp = mostRecentTimestamp.WallTime
 
 	// Set the stores map as the gossip persistent storage, so that
 	// gossip can bootstrap using the most recently persisted set of
 	// node addresses.
 	if err := n.storeCfg.Gossip.SetStorage(n.stores); err != nil {
-		return fmt.Errorf("failed to initialize the gossip interface: %s", err)
+		return fmt.Errorf("failed to initialize the gossip interface: %s", err), nil
 	}
 
-	// Bootstrap any uninitialized stores.
-	//
-	// TODO(tbg): address https://github.com/cockroachdb/cockroach/issues/39415.
-	// Should be easy enough. Writing the test is probably most of the work.
+	// Bootstrap any uninitialized stores and signal channel once complete.
+	bootstrapStoresComplete := make(chan struct{})
 	if len(state.newEngines) > 0 {
-		if err := n.bootstrapStores(ctx, state.firstStoreID, state.newEngines, n.stopper); err != nil {
-			return err
-		}
+		// We need to bootstrap additional stores asynchronously because we can't rely on
+		// the KV layer to be available yet. See:
+		//
+		// https://github.com/cockroachdb/cockroach/issues/39415
+		_ = n.stopper.RunAsyncTask(ctx, "bootstrap-stores", func(ctx context.Context) {
+			if err := n.bootstrapStores(ctx, state.firstStoreID, state.newEngines, n.stopper); err != nil {
+				log.Fatalf(ctx, "while bootstrapping additional stores: %v", err)
+			}
+			close(bootstrapStoresComplete)
+		})
+	} else {
```

Comment:
Add a comment here just spelling out that we're "done" seeing as how there aren't any new stores to be added.